pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -98,20 +112,41 @@
 
   public ControllerStarter(ControllerConf conf) {
     _config = conf;
-    _adminApp =
-        new ControllerAdminApiApplication(_config.getQueryConsoleWebappPath(), _config.getQueryConsoleUseHttps());
-    // Do not use this before the invocation of {@link PinotHelixResourceManager::start()}, which happens in {@link ControllerStarter::start()}
-    _helixResourceManager = new PinotHelixResourceManager(_config);
+    _controllerMode = conf.getControllerMode();
+    // Helix related settings.
+    _helixZkURL = HelixConfig.getAbsoluteZkPathForHelix(_config.getZkStr());
+    _helixClusterName = _config.getHelixClusterName();
+    _instanceId = conf.getControllerHost() + "_" + conf.getControllerPort();
+    _isUpdateStateModel = _config.isUpdateSegmentStateModel();
+    _enableBatchMessageMode = _config.getEnableBatchMessageMode();
+
     _metricsRegistry = new MetricsRegistry();
     _controllerMetrics = new ControllerMetrics(_metricsRegistry);
-    _executorService =
-        Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("restapi-multiget-thread-%d").build());
+    if (ControllerConf.ControllerMode.HELIX_ONLY.equals(_controllerMode)) {
```

Comment:
Prefer `if (_controllerMode == ControllerConf.ControllerMode.HELIX_ONLY)` for enum

Reply:
Done.